### ISSUE TITLE
Wrong RawForeignKeySpec.timing type

### ISSUE THREAD

**Issue by mattblang** [google/lovefield#166]
I noticed that `RawForeignKeySpec.timing` is set to `ConstraintAction`. It appears in [the tests](https://github.com/google/lovefield/blob/master/tests/schema/builder_test.js#L57) that it should be `ConstraintTiming`. I submitted [a PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9821) at DefinitelyTyped; I wanted to make an issue here to ensure that the fix in my PR is indeed correct.

**Comment by freshp86** [comment]
Responded in the PR. The canonical source of truth for our Typescript definitions is at https://github.com/google/lovefield/blob/master/dist/lovefield.d.ts#L212-217. Thanks for noticing the discrepancy.

**Comment by mattblang** [comment]
@freshp86 I saw [this article](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/), which seems to indicate that they are wanting to move away from any special type-fetching tool (like `typings`) altogether. It seems they have populated a `@types` npm package using type definitions currently in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped). In [this GitHub issue](https://github.com/Microsoft/TypeScript/issues/9184) one of the Microsoft employees says that `Publishing declaration files with your packages would be the preferred course of action`. The problem with the type definition file that is included in Lovefield is that it won't work when trying to use it directly (with the `files` tsconfig property) because of the `reference path` on `es6-promise`. I was able to use the `@types` package to grab what I assume is a "built" version of the types or something, but it is out of date (doesn't include the ConstraintTiming change I made). Based on the comment from that Microsoft employee, I guess packaging the definition file with your package is a preferred way to distribute your definition file, but I am not sure what processing occurred to produce a type file that does not `reference` another one.
